src/common/src/types/interval.rs

```diff
@@ -65,22 +66,96 @@ impl Interval {
         }
     }
 
-    pub fn get_days(&self) -> i32 {
-        self.days
+    /// Returns the total number of whole months.
+    ///
+    /// Note the difference between [`months`] and [`months_field`].
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "5 yrs 1 month".parse().unwrap();
+    /// assert_eq!(interval.months(), 61);
+    /// assert_eq!(interval.months_field(), 1);
+    /// ```
+    pub fn months(&self) -> i32 {
+        self.months
     }
 
-    pub fn get_months(&self) -> i32 {
-        self.months
+    /// Returns the number of days.
+    pub fn days(&self) -> i32 {
+        self.days
     }
 
-    pub fn get_usecs(&self) -> i64 {
+    /// Returns the total number of microseconds in a day.
+    pub fn usecs(&self) -> i64 {
         self.usecs
     }
 
-    pub fn get_usecs_of_day(&self) -> u64 {
+    /// Calculates the remaining number of microseconds in a day.
+    ///
+    /// Note the difference between [`usecs`] and [`usecs_of_day`].
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "-1:00:00".parse().unwrap();
+    /// assert_eq!(interval.usecs(), -1 * 60 * 60 * 1_000_000);
+    /// assert_eq!(interval.usecs_of_day(), 23 * 60 * 60 * 1_000_000);
+    /// ```
+    pub fn usecs_of_day(&self) -> u64 {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
+    /// Returns the years field.
+    pub fn years_field(&self) -> i32 {
+        self.months / 12
+    }
+
+    /// Returns the months field. range: 0-11
+    pub fn months_field(&self) -> i32 {
+        self.months % 12
+    }
+
+    /// Returns the days field.
+    pub fn days_field(&self) -> i32 {
+        self.days
+    }
+
+    /// Returns the hours field. range: -23..=23
+    pub fn hours_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 3600 % 24) as i32
+    }
```

Comment:
In PostgreSQL:
```
test=# select extract(hour from interval '100 hour');
 extract 
---------
     100
(1 row)
```

Basically, in `Interval`, the fields are (months, days, usecs), because days per month and hours per day can vary.

In `pg_itm` or the `_field` series, the separation is still preserved:
* `months` -> unlimited `years_field` and bounded `months_field`
* `days` -> `days` no change
* `usecs` -> unlimited `hours_field`, bounded `minutes_fields`, and bounded fractional seconds `seconds_in_micros`. I guess we agree the fractional seconds part should not be named as `_field`.